From c6aa80d830f6d37c864cb7d9d82919d16d0c4dbb Mon Sep 17 00:00:00 2001
From: "Mazlan, Hazwan Arif" <hazwan.arif.mazlan@intel.com>
Date: Fri, 22 Mar 2024 03:19:30 +0800
Subject: [PATCH 32/40] ui/gtk: Enable delay in gd configure while resize
 window

It makes more sense to call dpy_set_ui_info with delay enabled in gd_configure since it can be called too many times while user is resizing the window.

Signed-off-by: Dongwon, Kim <dongwon.kim@intel.com>
Rebased-by: Mazlan, Hazwan Arif <hazwan.arif.mazlan@intel.com>
---
 ui/gtk.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/ui/gtk.c b/ui/gtk.c
index b84ce9644e9c..b3bd7f3fcd37 100644
--- a/ui/gtk.c
+++ b/ui/gtk.c
@@ -2307,7 +2307,13 @@ static gboolean gd_configure(GtkWidget *widget,
     VirtualConsole *vc = opaque;
 
     if (vc->gfx.visible) {
-        gd_ui_show(vc);
+	QemuUIInfo info;
+	GdkWindow *window = gtk_widget_get_window(vc->gfx.drawing_area);
+
+	info = *dpy_get_ui_info(vc->gfx.dcl.con);
+	info.width = gdk_window_get_width(window);
+	info.height = gdk_window_get_height(window);
+	dpy_set_ui_info(vc->gfx.dcl.con, &info, true);
     }
 
     return FALSE;
-- 
2.34.1

